src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -713,6 +769,38 @@
     end
   end
 
+  describe 'POST #save_meta' do
+    before do
+      login user
+    end
+
+    context 'with an nonexistent project' do
+      let(:post_save_meta) { post :save_meta, params: { project: 'nonexistent_project' }, xhr: true }
+
+      it { expect{ post_save_meta }.to raise_error(Pundit::NotDefinedError) }
```

Comment:
lol fair point.. but I think I will do that in the next PR because there are already so many changes and failing specs in this one. And it would mean I have to change set_project_by_name so that might effect the behaviour of other actions which rely on it. is that ok?